lib/apollo/appWorkerLink.js

```diff
@@ -0,0 +1,120 @@
+import { ApolloLink, Observable } from 'apollo-link'
+import { SubscriptionClient } from 'subscriptions-transport-ws'
+import { API_WS_URL } from '../constants'
+
+export const hasSubscriptionOperation = ({ query: { definitions } }) => (
+  definitions.some(
+    ({ kind, operation }) =>
+      kind === 'OperationDefinition' && operation === 'subscription'
+  )
+)
+
+// WebSocket compliant interface to handle subscriptions via app worker
+class WorkerInterface {
+  constructor (url, protocol) {
+    this.url = url
+    this.protocol = protocol
+    this.readyState = WorkerInterface.OPEN
+  }
+
+  send (serializedMessage) {
+    window.postMessage(serializedMessage, '*')
+  }
+
+  set onmessage (fn) {
+    document.addEventListener('message', ({ data }) => {
+      const d = JSON.parse(data)
+
+      if (d.type === 'start' || d.type === 'data') {
```

Comment:
what about `stop`, `error` and `complete`? If error can happen with a specific query it should probably be handled